File: quadratic-algorithm/quadratic.go
package main

import (
	"fmt"
	"math"
)

/**
 * Quadratic - comparing every element to every other element
 * For each element check every element to see if it's the Sqrt of the current element
 * Time & Space Complexity - O(n^2) - We do O(n) work O(n) times - loop over list(n) "n" times
 * @param {[]int} list
 */
func quadratic(list []int) bool {
	result := false

	for i := 0; i < len(list); i++ {
		current := list[i]

		for j := 0; j < len(list); j++ {
			// Type conversion
			// math.Sqrt expects float64 and int provided
			calc := math.Sqrt(float64(list[j]))
			if int(calc) == current {
				result = true
			}
		}
	}

	return result
}

func main() {
	// true
	list := []int{3, 4, 5, 6, 7, 8, 16}
	fmt.Println(quadratic(list))

	// false
	l := []int{3, 4, 5, 6, 7, 8}
	fmt.Println(quadratic(l))
}

File: quadratic-algorithm/quadratic.js
/**
 * Quadratic - comparing every element to every other element
 * For each element check every element to see if it's the Sqrt of the current element
 * Time & Space Complexity - O(n^2) - We do O(n) work O(n) times - loop over list(n) "n" times
 * @param {array} list
 */
function quadratic(list) {
	let result = false;

	for (let i = 0; i < list.length; i++) {
		const current = list[i];

		for (let j = 0; j < list.length; j++) {
			if (Math.sqrt(list[j]) === current) {
				result = true;
			}
		}
	}

	return result;
}

// true
console.log(quadratic([3,4,5,6,7,8,16]));

// false
console.log(quadratic([3,4,5,6,7,8]));
